storage/pipeline/commit_batch.go

```diff
@@ -414,12 +415,20 @@ func (b *CommitBatcher) processIndividually(cfg sealiface.Config) ([]sealiface.C
 
 	var res []sealiface.CommitBatchRes
 
+	sectorsProcessed := 0
+
 	for sn, info := range b.todo {
 		r := sealiface.CommitBatchRes{
 			Sectors:       []abi.SectorNumber{sn},
 			FailedSectors: map[abi.SectorNumber]string{},
 		}
 
+		if cfg.MaxSectorProveCommitsSubmittedPerEpoch > 0 &&
+			uint64(sectorsProcessed) >= cfg.MaxSectorProveCommitsSubmittedPerEpoch {
+			time.Sleep(builtin.EpochDurationSeconds * time.Second)
+			sectorsProcessed = 0
```

Comment:
I'd try to avoid the sleep in any way possible.

Polling api.ChainHead every 2-3s, and looking for height change is probably good enough